# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerEvents.cpp
# Commit: f4d11e6a6e3f
# Full Hash: f4d11e6a6e3f74cd0f43c2a657b596b1eb4a1e55
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2020-03-26 15:34:27
# Regressor Bug: 1624810
# File Overlap Count: 1
# Description:
#   Bug 1624810 - Set the multi-zone JS holder flag on appropriate cycle collected classes r=mccr8
#   
#   Set the flag on the affected classes, which are:
#    - CallbackTimeoutHandler
#    - nsJSArgArray
# ==============================================================================

diff -r 4dcdeed2265f -r f4d11e6a6e3f dom/serviceworkers/ServiceWorkerEvents.cpp
--- a/dom/serviceworkers/ServiceWorkerEvents.cpp	Thu Mar 26 10:48:30 2020 +0000
+++ b/dom/serviceworkers/ServiceWorkerEvents.cpp	Thu Mar 26 10:49:06 2020 +0000
@@ -1253,7 +1253,7 @@
   aPorts = mPorts;
 }
 
-NS_IMPL_CYCLE_COLLECTION_CLASS(ExtendableMessageEvent)
+NS_IMPL_CYCLE_COLLECTION_MULTI_ZONE_JSHOLDER_CLASS(ExtendableMessageEvent)
 
 NS_IMPL_CYCLE_COLLECTION_UNLINK_BEGIN_INHERITED(ExtendableMessageEvent, Event)
   tmp->mData.setUndefined();